Improved accessibility of Quick Start checklist header

stevebaranski opened the pull request:
Fixes #11493. 

The `QuickStartChecklistHeader` is somewhat dynamic, so I devoted a bit of attention to updating the `accessibilityLabel` & `accessibilityHint` in conjunction with the view state.

_NB: It's probably best to review this in conjunction with #11578, as the same screen (and release notes entry) are impacted._

To test:
- Checkout the branch, verify it builds, and that tests pass.
- Replicate the steps in the issue with VoiceOver, and verify that the desired accessibility behavior is observed.

Update release notes:

- [X] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

etoledom commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistHeader.swift:
```diff
@@ -15,11 +16,29 @@ class QuickStartChecklistHeader: UIView {
+                hintToken = NSLocalizedString("Expands", comment: "This token is used with ~ to format the accessibility hint to reflect the expanded and collapsed state of the list.")
+            }
+
+            let accessibilityFormat = NSLocalizedString("%@ the list of completed tasks.", comment: "Accessibility hint for the list of completed tasks presented during Quick Start.")
```
In this case I believe that is better to localize the whole sentence. In other languages the order of words might be different, and the "Expands"/"Collapses" translation won't necessarily be always at the beginning.

etoledom commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistHeader.swift:
```diff
@@ -15,11 +16,29 @@ class QuickStartChecklistHeader: UIView {
             titleLabel.text = String(format: Constant.title, count)
+
+            // Admittedly pluralization support could be improved here. See also: `paaHJt-c3-p2`
+            let accessibilityFormat = NSLocalizedString("%i completed tasks, toggling expands & contracts the list of these tasks", comment: "Accessibility description for the list of completed tasks presented during Quick Start.")
```
Since this control holds a state, I think it would be good to state its current state in the accessibility label.
Something like: "Expanded. 1 Completed task. Button. (hint) expans & collapse the list of completed tasks"

etoledom replied:
It also might be helpful to let translators now that `%i` is a number (the count of completed tasks).

stevebaranski commented:
@etoledom thanks for the review, and for providing such helpful feedback. The app is better as a result. 😄 

I have pushed an additional commit that addresses your concerns. Because there are two different stimuli for the control (i.e., collapse & expand, as well as task completion), I created a new method that updates both labels in conjunction. It's now called when those values change, and via `awakeFromNib()`.

When you have a moment, would you be willing to take another look?

nheagy commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistHeader.swift:
```diff
@@ -1,6 +1,7 @@
 import Gridicons
 
 class QuickStartChecklistHeader: UIView {
+
```
why the space here?

stevebaranski replied:
This was unintentional - I'll tidy this up.

nheagy commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistHeader.swift:
```diff
@@ -60,8 +63,54 @@ class QuickStartChecklistHeader: UIView {
+
+        // From an accessibility perspective, this view is essentially monolithic, so we configure it accordingly
+        isAccessibilityElement = true
+        accessibilityTraits = [ .header, .button ]
```
We're not terribly consistent, but both our reference style guides have so spaces inside array list brackets: `[.header, .button]`

stevebaranski replied:
I'll tidy this up.

nheagy commented on WordPress/Classes/ViewRelated/Blog/QuickStartChecklistHeader.swift:
```diff
@@ -60,8 +63,54 @@ class QuickStartChecklistHeader: UIView {
+        updateCollapseHeaderAccessibility()
+    }
+
+    func updateCollapseHeaderAccessibility() {
```
I like the flow of this method.

stevebaranski commented:
Thanks for the input @nheagy - I pushed an additional commit that addresses those concerns.

@etoledom with @nheagy's approval and your reaction to my change comment, I'm going to infer that this is okay to merge. If I'm mistaken, please reach out, and I can address any concerns in a subsequent PR.
